packages/dx-grid-core/src/plugins/table-column-resizing/reducers.js

```diff
@@ -2,31 +2,32 @@ const MIN_SIZE = 40;
 
 export const changeTableColumnWidths = (state, { shifts }) => {
   const { columnWidths } = state;
-  const updatedColumnWidths = Object.keys(shifts)
-    .reduce((acc, columnName) => {
-      const size = Math.max(MIN_SIZE, columnWidths[columnName] + shifts[columnName]);
-      return Object.assign(acc, { [columnName]: size });
-    }, {});
+  const nextColumnWidth = columnWidths.slice();
+  let index;
+  const updatedColumn = nextColumnWidth.find((elem, elemIndex) => {
+    index = elemIndex;
+    return !!shifts[elem.columnName];
+  });
+  const size = Math.max(MIN_SIZE, updatedColumn.width + shifts[updatedColumn.columnName]);
+  nextColumnWidth.splice(index, 1, { columnName: updatedColumn.columnName, width: size });
+
   return {
     ...state,
-    columnWidths: { ...columnWidths, ...updatedColumnWidths },
-    draftColumnWidths: {},
+    columnWidths: nextColumnWidth,
+    draftColumnWidths: [],
   };
 };
 
 export const changeDraftTableColumnWidths = (state, { shifts }) => {
-  const { columnWidths, draftColumnWidths } = state;
-  const updatedDraftColumnWidths = Object.keys(shifts)
-    .reduce((acc, columnName) => {
-      if (shifts[columnName] === null) {
-        delete acc[columnName];
-        return acc;
-      }
-      const size = Math.max(MIN_SIZE, columnWidths[columnName] + shifts[columnName]);
-      return Object.assign(acc, { [columnName]: size });
-    }, Object.assign({}, draftColumnWidths));
+  const { columnWidths } = state;
+  const updatedColumn = columnWidths.find(elem => !!shifts[elem.columnName]);
```

Comment:
Should we also change shifts structure from an object to an array?

Reply:
I think that we want a one style of data